src/modules/editor/SoundEditor/SoundEditor.tsx

```diff
@@ -19,57 +80,164 @@ export class SoundEditor extends IEditor {
     render() {
-        const theme = useTheme()
-        const editor = new SoundEditor();
-        editor.sendData("test");
+        //const theme = useTheme()
         const cellWidth = 35;
         const cellHeight = 20;
         const gridWidth = 32;
-        const gridHeight = 16;
-        console.log("", cellWidth);
+        const gridHeight = 24;
         return (
-            <div
-                style={{
-                    backgroundColor: theme.colors.background,
-                    color: theme.colors.text,
-                    fontFamily: theme.typography.fontFamily,
-                    fontSize: theme.typography.fontSize,
-                }}
-            >
+            <div>
                 <div className="SoundEditor">
-                    <input
-                        type="number"
-                        min="0"
-                        max={10 - 1}
-                        value={1}
-
-                    />
-                    <div className="scrollable-container">
-
-                        <div className="grid"
-                            style={{
-                                gridTemplateColumns: `repeat(${gridWidth}, ${cellWidth}px)`,
-                                gridTemplateRows: `repeat(${gridHeight}, ${cellHeight}px)`,
-                            }}
-                        >
-                            {[...Array(gridHeight)].map((_, row) =>
-                                [...Array(gridWidth)].map((_, col) => {
-                                    const cellKey = `${row}-${col}`;
-                                    return (
-                                        <div
-                                            key={cellKey}
-                                            style={{
-                                                width: `${cellWidth}px`,
-                                                height: `${cellHeight}px`,
-                                                backgroundColor: "white",
-                                                boxSizing: "border-box"
-                                            }}
-                                        ></div>
-                                    );
-                                })
-                            )}
+
+                    <div className="editor-countainer">
+                        <div className="instrument-buttons">
+                            {this.getInstrumentButtons()}
                         </div>
+                        <div className="scrollable-container">
+                            <div
+                                className="grid"
+                                style={{
+                                    gridTemplateColumns: `repeat(${gridWidth}, ${cellWidth}px)`,
+                                    gridTemplateRows: `repeat(${gridHeight}, ${cellHeight}px)`,
+                                }}
+                            >
+                                {[...Array(gridHeight)].map((_, row) =>
```

Comment:
Be cautious with this construct in conjunction with `Array`. It can be quite expensive in both time and space, especially if there is a lot of cells to be made.